ako-gateway-api/status/gateway_status.go

```diff
@@ -140,31 +140,41 @@ func (o *gateway) Update(key string, option status.StatusOptions) {
 
 	status := gw.Status.DeepCopy()
 	addressType := gatewayv1.IPAddressType
-	status.Addresses = append(status.Addresses, gatewayv1.GatewayStatusAddress{
-		Type:  &addressType,
-		Value: option.Options.Vip[0],
-	})
-
-	// TODO: Add a way to propagate the error from the Rest layer to status layer.
-
+	if len(option.Options.Vip) != 0 {
+		status.Addresses = append(status.Addresses, gatewayv1.GatewayStatusAddress{
+			Type:  &addressType,
+			Value: option.Options.Vip[0],
+		})
+	}
 	condition := NewCondition()
-	condition.
-		Type(string(gatewayv1.GatewayConditionProgrammed)).
-		Status(metav1.ConditionTrue).
-		Reason(string(gatewayv1.GatewayReasonProgrammed)).
-		ObservedGeneration(gw.ObjectMeta.Generation).
-		Message("Virtual service configured/updated").
-		SetIn(&status.Conditions)
 
-	for i := range status.Listeners {
-		listenerCondition := NewCondition()
-		listenerCondition.
+	if option.Options.Message != "" {
+		condition.
+			Type(string(gatewayv1.GatewayConditionProgrammed)).
```

Comment:
Gateway Condition should be set to Accepted with gateway status false

Reply:
Changed gateway condition to accepted